torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp

```diff
@@ -343,7 +361,21 @@ void runPreAutodiffPassPipeline(std::shared_ptr<Graph>& graph) {
       "After CheckInplace (end of runPreAutodiffPassPipeline)\n", *graph);
 }
 
-void runNoGradOptimizations(std::shared_ptr<Graph>& graph) {
+FusionBehavior getCurrentBehavior(size_t remaining_depth) {
+  size_t curr_depth = 0;
+  auto curr_strategy = getFusionStrategy();
+  for (int i = static_cast<int>(curr_strategy.size()) -1; i >= 0; i--) {
+    curr_depth += curr_strategy[i].second;
+    if (remaining_depth <= curr_depth) {
+      return curr_strategy[i].first;
+    }
+  }
+  // should never get here
+  TORCH_WARN("Stratgy changed mid-invocation, NYI");
+  return FusionBehavior::DYNAMIC;
```

Comment:
we are changing the default behaviour here. I'd like to confirm we are okay with that.

Reply:
This should never get invoked, it means someone ran the fuser for a number of specializations, and then updated the global settings to a smaller number of allowed specializations.. I'm not sure what the behavior should be here. I'll change to STATIC conservatively